### ISSUE TITLE
QStyle::subElementRect() returns a rectangle with (wrongly) negative vertical position

### ISSUE THREAD

**Issue by sommerluk** [tsujan/Kvantum#676]
When calling QStyle::subElementRect(), it returns a rectangle with (wrongly) negative vertical position.

Example code:

```
// …
    const QRect myContentRectangle = style()->subElementRect( //
        QStyle::SE_LineEditContents,
        &myFrameStyleOption,
        this);
// …
```
Now, myContentRectangle is QRect(10,-98 420x415)

I would expect that its y() is not -98, but a positive value, probably similar to x().

(I’m using Ubuntu 21.04 which has kvantum 0.18.0+repack-1)

**Comment by tsujan** [comment]
What's the *real* problem?

**Comment by tsujan** [comment]
Oh, 0.18.0 is outdated.

**Comment by sommerluk** [comment]
The real problem is that I’m using the above code in a custom widget with a custom paintEvent.

Actual result (kvantum):

![actual](https://user-images.githubusercontent.com/6830724/127673435-60ad829f-301e-459d-8456-9d9fdc214407.png)


As you see, with kvantum, the content (the color palette) is not painted at the correct position, but moved upwards, so the upper part of the palette is not visible. This is because of the negative y() value that kvantum style returns in the above code.

Expected result (Breeze):

![expected](https://user-images.githubusercontent.com/6830724/127673461-8f613e4f-a940-47cf-a3d9-d95cf1fa4356.png)


Using this very same code works fine for Breeze, Fusion, ukui, Adwaita-dark, Windows, Adwaita, CDE, Motif, cleanlooks, gtk2, for which I have tested it.

**Comment by tsujan** [comment]
I understand and you're right. Since you're a programmer, I just need to ask you to check https://github.com/tsujan/Kvantum/blob/dd0487e6823301b537b2028a698ff95e710fcf4c/Kvantum/style/Kvantum.cpp#L14765 and compare it with 0.18.0.

Old versions remain as archives. The development is done in git; nothing is (or can be) backported. If you'd reported this when 0.17.0 was released, it would be fixed in 0.18.0. However, thanks for your attention!
